ucp/_libs/ucx_api.pyx

```diff
@@ -1443,3 +1561,196 @@ def stream_recv_nb(
+            raise ValueError(
+                "UCX is not configured with CUDA support, please add "
+                "`cuda_copy` and/or `cuda_ipc` to the UCX_TLS environment"
+                "variable and that the ucx-proc=*=gpu package is "
+                "installed. See "
+                "https://ucx-py.readthedocs.io/en/latest/install.html for "
+                "more information."
+            )
+        if not buffer._contiguous():
+            raise ValueError("Array must be C or F contiguous")
+
+        cdef ucp_request_param_t params
+        params.op_attr_mask = (
+            UCP_OP_ATTR_FIELD_CALLBACK |
+            UCP_OP_ATTR_FIELD_USER_DATA |
+            UCP_OP_ATTR_FIELD_FLAGS
+        )
+        params.cb.send = <ucp_send_nbx_callback_t>_send_nbx_callback
+        params.flags = UCP_AM_SEND_FLAG_REPLY
+        params.user_data = <void*>buffer.ptr
+
+        cdef ucs_status_ptr_t status = ucp_am_send_nbx(
+            ep._handle,
+            0,
+            NULL,
+            0,
+            <void*>buffer.ptr,
+            nbytes,
+            &params,
+        )
+        return _handle_status(
+            status, nbytes, cb_func, cb_args, cb_kwargs, name, ep._inflight_msgs
+        )
+
+    def am_recv_nb(
+        UCXEndpoint ep,
+        cb_func,
+        tuple cb_args=None,
+        dict cb_kwargs=None,
+        str name=None,
+    ):
+        """ This routine receives a message on a worker with the active message API.
+
+        TODO
+
+        Parameters
+        ----------
+        ep: UCXEndpoint
+            The endpoint that is used for the receive operation. Received active
+            messages are always targeted at a specific endpoint, therefore it is
+            imperative to specify the correct one here.
+        cb_func: callable
+            The call-back function, which must accept `request` and `exception` as the
+            first two arguments.
+        cb_args: tuple, optional
+            Extra arguments to the call-back function
+        cb_kwargs: dict, optional
+            Extra keyword arguments to the call-back function
+        name: str, optional
+            Descriptive name of the operation
+        """
+        worker = ep.worker
+
+        if cb_args is None:
+            cb_args = ()
+        if cb_kwargs is None:
+            cb_kwargs = {}
+        if name is None:
+            name = "am_recv_nb"
+        if Feature.AM not in worker._context._feature_flags:
+            raise ValueError("UCXContext must be created with `Feature.AM`")
+        cdef bint cuda_support
+
+        am_recv_pool = worker._am_recv_pool
+        ep_as_int = int(<uintptr_t>ep._handle)
+        if (
+            am_recv_pool is not None and
+            ep_as_int in am_recv_pool and
+            len(am_recv_pool[ep_as_int]) > 0
+        ):
+            recv_obj = am_recv_pool[ep_as_int].pop(0)
+            exception = recv_obj if issubclass(type(recv_obj), (Exception, )) else None
```

Comment:
```suggestion
            exception = recv_obj if isinstance(recv_obj, Exception) else None
```

Reply:
Nice catch, fixed in https://github.com/rapidsai/ucx-py/pull/701/commits/5cc8242b99fb4c4544ecc56d89908a72720ec134 .